### DISCUSSION TITLE
Best tts and vocoder models I found

### DISCUSSION THREAD

**Discussion by ShayBox** [coqui-ai/TTS#1891] [General]
I generated every combination of tts and vocoder model together, these are the resulting models I found with good combinations, though these still produce some bad combinations.

Here's a bash script
```bash
#!/usr/bin/env bash

declare -a text="The quick brown fox jumps over the lazy dog"

declare -a tts_models=(
    "tts_models/en/ljspeech/fast_pitch"
    "tts_models/en/ljspeech/glow-tts"
    "tts_models/en/ljspeech/speedy-speech"
    "tts_models/en/ljspeech/tacotron2-DDC_ph"
)

declare -a vocoder_models=(
    "vocoder_models/en/ljspeech/hifigan_v2"
    "vocoder_models/en/ljspeech/multiband-melgan"
    "vocoder_models/en/ljspeech/multiband-melgan"
    "vocoder_models/en/ljspeech/univnet"
    "vocoder_models/universal/libri-tts/fullband-melgan"
)

for model_name in "${tts_models[@]}"
do
    for vocoder_name in "${vocoder_models[@]}"
    do
        tts --text "${text}" --model_name "${model_name}" --vocoder_name "${vocoder_name}" --out_path "${model_name//\//-}-${vocoder_name//\//-}".wav
    done
done
```
Here's results, I removed any results that were just garbage sounds.  
Hopefully this helps someone like me just looking for good tts without having to train models  
[Samples.zip](https://github.com/coqui-ai/TTS/files/9388014/Samples.zip)

**Comment by MahdiEsrafili** [comment]
@ShayBox Thanks for your work. Can you please tell about processing time and hardware usage of these models?

**Reply by ShayBox** [reply]
to do a bulk it takes hours but for a single tts it takes longer for the module to init than to synth, its pretty quick, maybe not assistant speed but close

**Reply by MahdiEsrafili** [reply]
@ShayBox What about hardware usage? Did you consider memory usage? cpu or gpu?

**Reply by ShayBox** [reply]
I don't think it's that bad, but exact numbers I don't know, I have plenty and I'm running it on WSL Disrod so memory usage is already inflated, but if you're not training your own model it shouldn't use much to just synth the tts

**Reply by MahdiEsrafili** [reply]
Ok. If I train and use my own model, will it use more resources? why?

**Comment by ShayBox** [comment]
The best sounding in my opinion is `tts_models/en/ljspeech/fast_pitch` and `vocoder_models/en/ljspeech/hifigan_v2`
There's really no male voice models built-in

**Reply by gustaveracosta** [reply]
Omg, I was looking a male voice :(

**Reply by curious-broccoli** [reply]
@gustaveracosta tts_models/en/vctk/vits has very good male and female voices (multispeaker model)

**Reply by macaujack** [reply]
Thanks @codemmmmm. Now I find p230 to be the kind of male voice I want:
```
tts --text "$content" --model_name "tts_models/en/vctk/vits" --out_path "$out" --speaker_idx p230
```

**Reply by cw72** [reply]
@macaujack How you run that code? Everytime I try to run a model without a default_vocoder like "tts_models/en/vctk/vits", I get an error: No espeak backend found. Install espeak-ng or espeak to your system.

**Reply by curious-broccoli** [reply]
try installing espeak?

**Reply by skwzrd** [reply]
Yes, `apt install espeak-ng` will solve it

**Comment by curious-broccoli** [comment]
Have you tried the VITS models (they also don't use vocoder models I think)?

**Reply by ShayBox** [reply]
If it's in the default list then yes, otherwise no

**Comment by knochenhans** [comment]
Hi, some of the `tts_models/en/vctk/vits` multispeaker indexes also sound pretty good and have a nice British accent. I’ve been using `p273`, `p330`, and `p234` quite a lot.

**Reply by ShayBox** [reply]
Ah that's why I didn't test them, I have no idea how to use the multispeaker thing, the readme doesn't explain much very well on how to use the cli

**Reply by knochenhans** [reply]
You just have to give it the speaker index, like so: `tts --model_name tts_models/en/vctk/vits --speaker_idx p363 --text "this is a test"`. The available speakers can be listed with this: `tts --model_name tts_models/en/vctk/vits --list_speaker_idxs`.

**Reply by pncnmnp** [reply]
A follow up on the above comment - 
In addition to the ones mentioned by @knochenhans, all of these are worth looking into - `(226, 227, 229, 230, 239, 240, 241, 245, 248, 260, 273, 267, 285, 306, 336)`.
Out of these, the ones I loved were `241, 267, 287, 336`. `241` is nice but a bit too fast. `267` has a really nice deep voice, which would sound good for conversations. `287` also has a deep voice.

**Reply by prakharjain-Ae0** [reply]
347 335 273 <-- my favs. 273 is good

**Reply by mikebilly** [reply]
are there any good non-british english male voices?

**Reply by pncnmnp** [reply]
@mikebilly I don't think that's specified in any doc. I recommend you to fire up `tts-server` (something like `tts-server --model_name tts_models/en/vctk/vits`) and query all the speakers.

Edit: You can even write a python script to quickly generate different voices. [Something like this](https://github.com/pncnmnp/phoenix10.1/blob/master/radio.py#L666).

**Reply by xenotropic** [reply]
They are specified in the corpus which has a speaker_ids file, but in coqui they got scrambled, see #2258. At least as of three months ago in January when I ran a script to generate all the voices, these were American accents in Coqui's VCTK-VITS:  256 M, 257 F, 270 F, 287 M, 293 F, 317 M, 360 F (may not be all)

**Reply by knoopx** [reply]
tried many of them, impressed with most them, wasn't expecting great results!

**Comment by zzzgit** [comment]
`tts_models/en/ljspeech/vits--neon` is the best

**Reply by devasheeshG** [reply]
please try jenny_neural

**Reply by hubertlepicki** [reply]
@devasheeshG is it the same as "tts_models/en/jenny/jenny" ? I don't see jenny_neural when I --list_models, only the other one

**Reply by devasheeshG** [reply]
Yup

**Reply by hubertlepicki** [reply]
Thanks, I tested it, it's very nice (although sounds artificial, but that's not exactly bad in my use case).

**Comment by omega3** [comment]
What is currently the best British English voice? Can anyone share their trained model? I have pretty week Intel G4560 with no GPU, so I have no chances at the moment to train.

**Comment by codepharmer** [comment]
> I generated every combination of tts and vocoder model together, these are the resulting models I found with good combinations, though these still produce some bad combinations.
> 
> Here's a bash script
> ```bash
> #!/usr/bin/env bash
> 
> declare -a text="The quick brown fox jumps over the lazy dog"
> 
> declare -a tts_models=(
>     "tts_models/en/ljspeech/fast_pitch"
>     "tts_models/en/ljspeech/glow-tts"
>     "tts_models/en/ljspeech/speedy-speech"
>     "tts_models/en/ljspeech/tacotron2-DDC_ph"
> )
> 
> declare -a vocoder_models=(
>     "vocoder_models/en/ljspeech/hifigan_v2"
>     "vocoder_models/en/ljspeech/multiband-melgan"
>     "vocoder_models/en/ljspeech/multiband-melgan"
>     "vocoder_models/en/ljspeech/univnet"
>     "vocoder_models/universal/libri-tts/fullband-melgan"
> )
> 
> for model_name in "${tts_models[@]}"
> do
>     for vocoder_name in "${vocoder_models[@]}"
>     do
>         tts --text "${text}" --model_name "${model_name}" --vocoder_name "${vocoder_name}" --out_path "${model_name//\//-}-${vocoder_name//\//-}".wav
>     done
> done
> ```
> Here's results, I removed any results that were just garbage sounds.  
> Hopefully this helps someone like me just looking for good tts without having to train models  
> [Samples.zip](https://github.com/coqui-ai/TTS/files/9388014/Samples.zip)

This is great! Thanks for sharing 😊
